parsers/US_SPP.py

```diff
@@ -420,6 +430,97 @@ def fetch_exchange(
     return exchanges
 
 
+@refetch_frequency(timedelta(minutes=30))
+def fetch_realtime_local_marginal_price(
+    zone_key: ZoneKey,
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list[dict[str, Any]]:
+    if target_datetime is None:
+        target_datetime = datetime.now(tz=timezone.utc)
+    if target_datetime.tzinfo is None:
+        target_datetime = target_datetime.replace(tzinfo=timezone.utc)
+
+    prices = PriceList(logger)
+
+    # Get data for target datetime and previous 30 minutes in 5 min intervals
+    for minutes in range(0, 35, 5):
+        check_datetime = target_datetime - timedelta(minutes=minutes)
+        url = get_realtime_url(check_datetime)
+        raw_data = get_data(url, session)
+
+        spp_data = raw_data[raw_data["Settlement Location"] == "SPPNORTH_HUB"]
+        for _, row in spp_data.iterrows():
+            prices.append(
+                zoneKey=zone_key,
+                datetime=datetime.strptime(
+                    row["GMTIntervalEnd"], "%m/%d/%Y %H:%M:%S"
+                ).replace(tzinfo=timezone.utc),
+                price=row["LMP"],
+                currency="USD",
+                source=SOURCE,
+            )
+
+    price_list = prices.to_list()
+    return price_list
+
+
+@refetch_frequency(timedelta(days=1))
+def fetch_dayahead_local_marginal_price(
+    zone_key: ZoneKey,
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    if target_datetime is None:
+        target_datetime = datetime.now(tz=timezone.utc)
+
+    url = get_dayahead_url(target_datetime)
+    raw_data = get_data(url, session)
+    # filter by column "Settlement Location" so it only includes SPPNORTH_HUB
+    spp_data = raw_data[raw_data["Settlement Location"] == "SPPNORTH_HUB"]
+    prices = PriceList(logger)
+    for _, row in spp_data.iterrows():
+        prices.append(
+            zoneKey=zone_key,
+            datetime=datetime.strptime(
+                row["GMTIntervalEnd"], "%m/%d/%Y %H:%M:%S"
+            ).replace(tzinfo=timezone.utc),
+            price=row["LMP"],
+            currency="USD",
+            source=SOURCE,
+        )
+
+    price_list = prices.to_list()
+    return price_list
+
+
+def get_closest_5_minutes_datetime(datetime: datetime) -> datetime:
```

Comment:
The variable here is shadowing the datetime built in module, this makes it so you can't use the built in module and might have unexpected results.

I would advice against this and simply rename it to target_datetime or something similar.